iohk/UpdateProposal.hs

```diff
@@ -437,36 +434,18 @@ updateProposalFindInstallers opts env = do
   void $ doCheckConfig params
   echo "*** Finding installers"
   let rev = unGitRevision . cfgDaedalusRevision $ params
-  res <- using $ realFindInstallers rev (configurationKeys env)
-  echo "*** Finished. Moving files to work dir"
-  res' <- moveInstallersToWorkDir opts res
-  writeWikiRecord opts res'
-  storeParams opts (UpdateProposalConfig2 params res')
-
--- | Move installer files from wherever they were found into the work dir.
-moveInstallersToWorkDir :: CommandOptions -> InstallersResults -> Shell InstallersResults
-moveInstallersToWorkDir opts InstallersResults{..} = do
-  bk <- case buildkiteResult of
-    Just res -> do
-      p <- move (bkLocalPath res)
-      pure . Just $ res { bkLocalPath = p }
-    Nothing -> pure Nothing
-  av <- case appveyorResult of
-    Just res -> do
-      p <- move (avLocalPath res)
-      pure . Just $ res { avLocalPath = p }
-    Nothing -> pure Nothing
-  pure $ InstallersResults
-    { buildkiteResult = bk
-    , appveyorResult = av
-    , globalResult = globalResult }
-  where
-    move :: Text -> Shell Text
-    move src = do
-      let src' = fromText src
-      let dst = installersPath opts (filename src')
-      mv src' dst
-      pure (format fp dst)
+      destDir = Just (installersDir opts)
+  res <- liftIO $ realFindInstallers (configurationKeys env) (installerForEnv env) rev destDir
+  echo "*** Finished."
+  writeWikiRecord opts res
+  storeParams opts (UpdateProposalConfig2 params res)
+
+installerForEnv :: Environment -> CIResult -> Bool
+installerForEnv env = matchNet . installerNetwork . ciResultLocalPath
+  where matchNet n = case env of
+          Production -> n == Just InstallerMainnet
+          Staging    -> n == Just InstallerStaging
+          _          -> True
```

Comment:
Thanks, I have added a comment to try and clarify this function.

I think it's OK because `env` comes from the `--config` option. So if `mainnet.yaml` then `env` is `Production`, if `staging-testnet.yaml` then `env` is `Staging`.

So what it's saying is that when running under the production config, proposing an update for the mainnet, it will only find installers containing "mainnet" in their filename. When running under a development config, any installer will do, just so that it's possible to test this script on a development network.

There is a fourth option of the `Environment` type which is `Any` and I think this should be removed at some point, but I didn't do it today.